packages/slice-machine/pages/onboarding.tsx

```diff
@@ -140,6 +140,40 @@ const StepIndicator = ({
   );
 };
 
+function handleTracking(props: {
+  step: number;
+  startTime: number;
+  maxSteps: number;
+}) {
+  const state = useRef(props);
+
+  useEffect(() => {
+    state.current = props;
+  }, [props]);
+
+  useEffect(
```

Comment:
The Goal of this PR is to track the end of the onboarding.
Why did you make a useEffect that handles himself instead of adding something to the existing functions that skip / end the onboarding ?

Reply:
I think useEffect are uncontrolled and not very stable compare to a skip button trigering once the call.

Reply:
onunmount is called in more cases than pressing the skip button

they could close the screen or find another way to exit, like changing the url.